File: Valghalla.Internal.Web/src/features/administration/election/models/create-election-request.ts
export interface CreateElectionTaskTypeCommunicationTemplateRequest {
  taskTypeId: string;
  confirmationOfRegistrationCommunicationTemplateId?: string;
  confirmationOfCancellationCommunicationTemplateId?: string;
  invitationCommunicationTemplateId?: string;
  invitationReminderCommunicationTemplateId?: string;
  taskReminderCommunicationTemplateId?: string;
  retractedInvitationCommunicationTemplateId?: string;
  removedFromTaskCommunicationTemplateId?: string;
  removedByValidationCommunicationTemplateId?: string;
}

export interface CreateElectionRequest {
  title: string;
  electionTypeId: string;
  lockPeriod: number;
  electionStartDate: Date;
  electionEndDate: Date;
  electionDate: Date;
  workLocationIds: string[];
  confirmationOfRegistrationCommunicationTemplateId?: string;
  confirmationOfCancellationCommunicationTemplateId?: string;
  invitationCommunicationTemplateId?: string;
  invitationReminderCommunicationTemplateId?: string;
  taskReminderCommunicationTemplateId?: string;
  retractedInvitationCommunicationTemplateId?: string;
  removedFromTaskCommunicationTemplateId?: string;
  removedByValidationCommunicationTemplateId?: string;
  electionTaskTypeCommunicationTemplates?: Array<CreateElectionTaskTypeCommunicationTemplateRequest>;
}

File: Valghalla.Internal.API/Requests/Administration/Election/CreateElectionRequest.cs
﻿using Valghalla.Internal.Application.Modules.Administration.Election.Requests;

namespace Valghalla.Internal.API.Requests.Administration.Election
{
    public sealed record CreateElectionRequest
    {
        public string Title { get; init; } = string.Empty;
        public Guid ElectionTypeId { get; init; }
        public int LockPeriod { get; init; }
        public DateTime ElectionStartDate { get; init; }
        public DateTime ElectionEndDate { get; init; }
        public DateTime ElectionDate { get; init; }
        public Guid? ConfirmationOfRegistrationCommunicationTemplateId { get; set; }
        public Guid? ConfirmationOfCancellationCommunicationTemplateId { get; set; }
        public Guid? InvitationCommunicationTemplateId { get; set; }
        public Guid? InvitationReminderCommunicationTemplateId { get; set; }
        public Guid? TaskReminderCommunicationTemplateId { get; set; }
        public Guid? RetractedInvitationCommunicationTemplateId { get; set; }
        public IEnumerable<Guid> WorkLocationIds { get; init; } = Array.Empty<Guid>();
        public IEnumerable<CreateElectionTaskTypeCommunicationTemplateRequest> ElectionTaskTypeCommunicationTemplates { get; init; } = Array.Empty<CreateElectionTaskTypeCommunicationTemplateRequest>();
    }
}
